src/main/scala/org/deepdive/ddlog/DeepDiveLogSemanticChecker.scala

```diff
@@ -156,6 +157,46 @@ object DeepDiveLogSemanticChecker extends DeepDiveLogHandler {
     }
   }
 
+  def checkSupervisionLableType(s: ExtractionRule, expType: VariableType, SupVariable: VarPattern, b: BodyAtom) {
+    b.terms.zipWithIndex.foreach {
+      case (SupVariable, index) => {
+        val expColumnType = if (expType.isInstanceOf[BooleanType.type]) "boolean" else "text"
+        if (schemaDeclaration(b.name).a.types(index).toLowerCase != expColumnType) {
+          error(s, s"Unexpected supervision type")
```

Comment:
Can we show a more precise error message, such as `s"Supervision column ${SupVariable} should be ${expColumnType} type but is ${actualColumnType} type"`?